db/background_mgr.go

```diff
@@ -164,6 +347,40 @@ func (b *BackgroundManager) SetError(err error) {
 
 	b.lastError = err
 	b.State = BackgroundProcessStateError
+	b.Terminate()
+}
+
+// UpdateStatusClusterAware gets the current local status from the running process and updates the status document in
+// the bucket. Implements a retry. Used for Cluster Aware operations
+func (b *BackgroundManager) UpdateStatusClusterAware() error {
+	err, _ := base.RetryLoop("UpdateStatusClusterAware", func() (shouldRetry bool, err error, value interface{}) {
+		status, err := b.getStatusLocal()
+		if err != nil {
+			return true, err, nil
+		}
+
+		err = b.clusterAwareOptions.bucket.SetRaw(b.clusterAwareOptions.StatusDocID(), 0, status)
+		if err != nil {
+			return true, err, nil
+		}
+
+		return false, nil, nil
+	}, base.CreateSleeperFunc(5, 100))
+	return err
+}
+
+// UpdateHeartbeatDocClusterAware simply performs a touch operation on the heartbeat document to update its expiry.
+// Implements a retry. Used for Cluster Aware operations
+func (b *BackgroundManager) UpdateHeartbeatDocClusterAware() error {
+	err, _ := base.RetryLoop("", func() (shouldRetry bool, err error, value interface{}) {
+		_, err = b.clusterAwareOptions.bucket.Touch(b.clusterAwareOptions.HeartbeatDocID(), 30)
```

Comment:
30 seconds seems like a long time to detect node failure, given that we're attempting to perform a touch every second.  What's the rationale here?

Reply:
It was for the case where we allow a failure to be ignored until the next interval but with the retry this can be lowered.